python/ray/serve/scripts.py

```diff
@@ -284,70 +287,51 @@ def run(
     blocking: bool,
     gradio: bool,
 ):
-    sys.path.insert(0, app_dir)
+    # If no local ray instance is running, start one.
+    if address is None and not ray.is_initialized():
+        ray.init(namespace=SERVE_NAMESPACE)
 
     final_runtime_env = parse_runtime_env_args(
         runtime_env=runtime_env,
         runtime_env_json=runtime_env_json,
         working_dir=working_dir,
     )
+    if "env_vars" not in final_runtime_env:
+        final_runtime_env["env_vars"] = {}
+    final_runtime_env["env_vars"]["RAY_JOB_STOP_SIGNAL"] = "SIGINT"
+
+    # The job to run on the cluster, which imports and runs the serve app.
+    from ray.serve import run_script
+
+    # Use Ray Job Submission to run serve.
+    client = JobSubmissionClient(address)
+    submission_id = client.submit_job(
+        entrypoint=(
+            f"python {run_script.__file__} "
```

Comment:
You're right, I missed that! This becomes an issue for remote clusters. I am not sure how to start a remote cluster in unit tests, can you point me to the right place?

Reply:
Good question, we don't have a way to start remote clusters in unit tests...  The question would be how to simulate a local file not being present on a "remote" cluster, one way we've done this in the past is to just delete the file (https://github.com/ray-project/ray/blob/9cd4dd77495e2d5481db1face441016eba9a06fe/python/ray/tests/test_runtime_env_complicated.py#L665-L669) but in our case, we don't want to delete `run_script.py` since it's part of Ray. One option would be to rename the file after the working_dir is done being uploaded but before the job executes, and then write a `pytest` fixture to name it back to `run_script.py` when the test exits, but I'm not sure how to time the first renaming.

My next thought would be to add a simple release test to cover this (since release tests do use a remote cluster), but I'm realizing now that's also not supported: Right now the release test infrastructure just lets you specify a command to be executed on the remote cluster itself, so we'd only be able to test running `serve run` on the cluster (and all the nodes here happen to have the same filesystem layout, so the previous approach with the absolute filepath would still spuriously pass.) 

I think we can skip adding this test for now. I'll open a feature request for the release test infra to allow adding tests that run commands on the client machine, not the remote cluster.